daemon/container_operations_unix.go

```diff
@@ -344,6 +345,16 @@ func killProcessDirectly(cntr *container.Container) error {
 	if status.Err() != nil {
 		// Ensure that we don't kill ourselves
 		if pid := cntr.GetPID(); pid != 0 {
+
+			// since we can not kill a zombie pid, add zombie check here
+			isZombie, err := system.IsProcessZombie(pid)
```

Comment:
Should we move this block after the `unix.Kill(pid, 9)` block?

Reply:
yes, I have thought about that, I think It is rare to get this `killProcessDirectly` executed when it happens the container Pid might have been abnormal, So do a precheck for the Pid might be more efficiently, not sure that is right please correct me if  I'm wrong~

Reply:
We thought doing the zombie check was rather heavy for that, so better to first try the regular kill and only for the exceptional case there's a zombie process to do the check

Reply:
I concur -- this check is heavy and should be done as a last resort (such as when after `kill -9` the process is still there)

Reply:
execution order changed.

Reply:
hmm...  IT looks more naturally add check after the kill